### PR TITLE
Add hints for clickhouse client options

### PR DESCRIPTION
**Pull request by lingtaolf**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

Detailed description / Documentation draft:
This PR solved the problem mentioned in #32237  
  
After changing, client will give the hints if you used a wrong option:  
```text   
(base) ➜  ~ clickhouse-client --hardware_utilization 1
Code: 552. DB::Exception: Unrecognized option '--hardware_utilization'. Maybe you meant ['--hardware-utilization']. (UNRECOGNIZED_ARGUMENTS)
```

### CODE REVIEW

**Comment by kssenii** [comment]
Can you please add functional test?

File: src/Client/ClientBase.cpp [hunk 1]
```diff
@@ -1713,6 +1719,17 @@ void ClientBase::init(int argc, char ** argv)
...
     addOptions(options_description);
+
+    auto getter = [](const auto & op) { 
+        String op_long_name = op->long_name();
+        return "--" + String(op_long_name); 
+    };
+
+    const auto & main_options = options_description.main_description->options();
+    const auto & external_options = options_description.external_description->options();
+    std::transform(main_options.begin(),main_options.end(), std::back_inserter(cmd_options), getter);
```
Comment (kssenii) [inline src/Client/ClientBase.cpp:1730]:
```suggestion
    std::transform(main_options.begin(), main_options.end(), std::back_inserter(cmd_options), getter);
```

**Comment by lingtaolf** [comment]
> Can you please add functional test?

I will try !💪🏻

**Comment by cmsxbc** [comment]
Something maybe off topic, but it's better to use `git merge`  in command line or at mergifyio in command for merging mastger commit...
what you have done make your pr commit and changes unreadable....

Mergifyio usage example: https://github.com/ClickHouse/ClickHouse/pull/32291#issuecomment-991866597

**Comment by lingtaolf** [comment]
> Something maybe off topic, but it's better to use `git merge` in command line or at mergifyio in comment for merging mastger commit... what you have done make your pr commit and changes unreadable....and what's more important, cause unnecessary confilics.
> 
> Mergifyio usage example: [#32291 (comment)](https://github.com/ClickHouse/ClickHouse/pull/32291#issuecomment-991866597)

@cmsxbc Thanks for you advices. There must be some thing wrong when I rebasing the master branch. I'm trying to fix and rebase the commits.

**Comment by lingtaolf** [comment]
Sry, there are so many conflicts to solve. I think it's better to create anther PR .  #32841

**Comment by lingtaolf** [comment]
Sry, There're so many conflicts need to solve. I think it's better to create another PR. #32841

**Pull request closed without merging**